sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as log from "./log";
+import { EventProcessorOptions, PartitionProcessor } from "./eventProcessor";
+import { PartitionContext } from "./partitionContext";
+import { EventHubClient } from "./eventHubClient";
+import { EventPosition } from "./eventPosition";
+import { EventHubConsumer } from "./receiver";
+
+export class PartitionPump {
+  private _partitionContext: PartitionContext;
+  private _eventHubClient: EventHubClient;
+  private _partitionProcessor: PartitionProcessor;
+  private _processorOptions: EventProcessorOptions;
+  private _receiver: EventHubConsumer | undefined;
+  private _isReceiving: boolean = false;
+
+  constructor(
+    eventHubClient: EventHubClient,
+    partitionContext: PartitionContext,
+    partitionProcessor: PartitionProcessor,
+    options?: EventProcessorOptions
+  ) {
+    if (!options) options = {};
+    this._eventHubClient = eventHubClient;
+    this._partitionContext = partitionContext;
+    this._partitionProcessor = partitionProcessor;
+    this._processorOptions = options;
+  }
+
+  async start(partitionId: string): Promise<void> {
+    await this._partitionProcessor.initialize!();
+    await this._receiveEvents(partitionId);
```

Comment:
I think you actually don't want to await this function call because it will potentially keep running until `stopped` is called. The log message won't fire until it's no longer receiving :)